src/driver/op.rs

```diff
@@ -40,11 +40,19 @@ pub(crate) struct Op<T: 'static, CqeType = SingleCQE> {
 /// A Marker for Ops which expect only a single completion event
 pub(crate) struct SingleCQE;
 
+/// A Marker for Operations will process multiple completion events,
+/// which combined resolve to a single Future value
+pub(crate) struct MultiCQEFuture;
+
 pub(crate) trait Completable {
     type Output;
     fn complete(self, cqe: CqeResult) -> Self::Output;
 }
 
+pub(crate) trait Updateable: Completable {
```

Comment:
I believe a comment is warranted for future designers. Something to the effect that the 'update' in this Trait, when used in the MultiCQEFuture poll, will be called when the future is polled, on all 'more' entries found in the completion list, draining the list in the one poll cycle.

Because another design choice could have been to have 'update' called when the driver read another cqe from the cqueue and not keep the slab linked list at all. ... Well, come to think of it, I'm not sure the driver's complete would have access to the 'update', maybe it doesn't know the type of the op.

Not to mention some other op implementations, like streams, would want each 'more' cqe sent back as the result of the poll, so the completion list wouldn't be drained immediately.

So maybe this is just me thinking out loud again.

Reply:
How about

// `update` is called for the intermediate results when the future is polled. Before being polled, the intermediate results build up in a completions list.